src/main/java/hudson/plugins/deploy/tomcat/TomcatAdapter.java

```diff
@@ -28,18 +33,24 @@
      * Top URL of Tomcat.
      */
     public final String url;
-    private final String path;
 
-    public TomcatAdapter(String url, String credentialsId) {
-        super(credentialsId);
-        this.url = url;
-        this.path = null;
-    }
+    /**
+     * Alternative context that override context defined in plugin main configuration
+     */
+    public final DeploymentContext alternativeDeploymentContext;
```

Comment:
This seems overkill. Please keep changes to a minimum (a final String is IMHO enough in this case).